### DISCUSSION TITLE
Uncontracted Lorentz indices

### DISCUSSION THREAD

**Discussion by hongyi18** [FeynCalc/feyncalc#203] [Q&A] [answered]
I notice that in the development version, a free Lorentz index is always understood as an upper index, according to [FeynCalc manual (development version)](https://feyncalc.github.io/referenceDev) Is this also true in the stable version 9.3.1?

**Comment by vsht** [comment, marked as answer]
Yes, although it actually would matter only when breaking manifest Lorentz covariance via `LorentzToCartesian`. Do you have some particle case in mind?

**Reply by hongyi18** [reply]
Thanks for your answer. I am dealing with traces of gamma matrix products, with expressions like (you may directly copy and run in Mathematica)
```
SP[p1] = me^2;
SP[p2] = me^2;
SP[q] = 0;

\[Beta] := 1/(SP[p1-q]-me^2);
\[Delta] := 1/(SP[p2+q]-me^2);
Mat1 := \[Beta] GA[0] . (GS[p1-q]+me) . GA[5] + \[Delta] GA[5] . (GS[p2+q]+me) . GA[0];
Mat2 := \[Beta] GA[5] . (GS[p1-q]+me) . GA[0] + \[Delta] GA[0] . (GS[p2+q]+me) . GA[5];  (* \[Gamma]^0 Mat1^\[Dagger] \[Gamma]^0 *)

-DiracTrace[Mat1 . (GS[p1]+me) . Mat2 . (GS[p2]+me)] //DiracSimplify;
MSq = % //FCI
```
The final expression contains the temporal components of 4 momenta, and I just want to make sure that I understand these temporal components correctly.

**Reply by vsht** [reply]
That should be totally fine, all temporal indices are always upper indices. BTW, you can actually assign some values to the temporal components just the way you do it for scalar products,
e.g. via
```
TC[q] = q0
TC[p1] = p10
```

**Reply by hongyi18** [reply]
That's neat. Thanks!
